docs/adr/0038-use-component-library.md

```diff
@@ -0,0 +1,59 @@
+# Use a React component library
+
+- Status: proposed
+- Deciders: Reino Muhl
+- Date: 2023-04-03
+
+## Context and Problem Statement
+
+As part of a major refactor of the Subscription Platform, the team is trying to simplify and reduce the number of components used by the Subscription Platform web app. To that end the team is investigating whether or not to replace some of the more commonly used components, such as a Tooltip, DialogModal, etc., with a component library that provides out of the box features such as accessibility and screen reader support, for example.
+
+Additionally, maintaining and implementing the existing components sometimes comes with additional developer cost. For example, during a recent refactor introducing TailwindCSS, it was difficult to refactor items in the bespoke component library due to deeply nested styling.
+
+## Decision Drivers
+
+- Accessibility
+- Speed of development
+- Maintainability
+
+## Considered Options
+
+- A. Unstyled component library
+- B. Styled component library
+- C. Bespoke components
+
+## Decision Outcome
+
+Chosen option: A. Unstyled component library. Using an unstyled component library, also sometimes referred to as a headless component library, meets all key decision drivers, with few draw backs. Unstyled components allows the team to bring their own existing styling library, TailwindCSS, and brings
+Unstyled components work well with styling libraries like TailwindCSS, and provide accessibility features etc. out of the box, which would otherwise need to be added and maintained by the team, on bespoke components.
+
+## Pros and Cons of the Options
+
+### A. Unstyled component library
+
+- Good, because it provides accessibility, screen reader support, keyboard navigation, focus management, etc. out of the box
+- Good, because it provides a consistent developer experience
+- Good, because it allows the team to bring the existing design system
+- Good, because it works well with the teams styling library TailwindCSS
+- Good, because it can be dropped in where necessary, without interferring with existing components
+- Neutral, because it doesn't have all components and still requires developer time to add new components, or wait for the library to add support
+- Bad, because it is opinionated
+- Bad, because it adds additional dependencies
+
+### B. Styled component library
```

Comment:
@pdehaan mentioned this styled component library in another PR, thought I'd note it here because I had not heard of it before, and it's tailwind-based:
https://flowbite.com/

Reply:
Ah nice, that looks interesting. I also hadn't heard of it before. Thank you for sharing!

Reply:
I think @LZoog also mentioned Headless UI above in https://github.com/mozilla/fxa/pull/15124#discussion_r1175436084
Plus there is [a paid] https://tailwindui.com/components, which claims:
> _"Over 500+ professionally designed, fully responsive, expertly crafted component examples you can drop into your Tailwind projects and customize to your heart’s content."_

Although looking at the TailwindUI code (if you have a paid acct, I have a personal acct if you want to take a look over Zoom) they use the Headless UI component set and give you a lot of React boilerplate code:

```jsx
import { useState } from 'react'
import { Dialog } from '@headlessui/react'
import { Bars3Icon, XMarkIcon } from '@heroicons/react/24/outline'
…
```

Plus a few more I hadn't heard of yet in https://prismic.io/blog/tailwind-component-library.